test-integration/src/test/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystemTestHelper.java

```diff
@@ -56,38 +60,40 @@ public AmazonS3FileSystemTestHelper(String accessKey, String secretKey, boolean
 	}
 
 	@Override
-	@Before
+	@BeforeAll
 	public void setUp() throws ConfigurationException, IOException, FileSystemException {
 		open();
-		s3Client.createBucket(bucketName);
+		if (!s3Client.doesBucketExist(bucketName)) {
+			s3Client.createBucket(bucketName);
+		}
 	}
 
 	@Override
-	@After
+	@AfterEach
 	public void tearDown() throws Exception {
 		cleanUpBucketAndShutDown(s3Client);
 	}
 	
 	public void cleanUpBucketAndShutDown(AmazonS3 s3Client) {
 		if(s3Client.doesBucketExistV2(bucketName)) {
-			 ObjectListing objectListing = s3Client.listObjects(bucketName);
-	            while (true) {
-	                Iterator<S3ObjectSummary> objIter = objectListing.getObjectSummaries().iterator();
-	                while (objIter.hasNext()) {
-	                    s3Client.deleteObject(bucketName, objIter.next().getKey());
-	                }
-	    
-	                // If the bucket contains many objects, the listObjects() call
-	                // might not return all of the objects in the first listing. Check to
-	                // see whether the listing was truncated. If so, retrieve the next page of objects 
-	                // and delete them.
-	                if (objectListing.isTruncated()) {
-	                    objectListing = s3Client.listNextBatchOfObjects(objectListing);
-	                } else {
-	                    break;
-	                }
-	            }
-			s3Client.deleteBucket(bucketName);
+			ObjectListing objectListing = s3Client.listObjects(bucketName);
+			while (true) {
+				Iterator<S3ObjectSummary> objIter = objectListing.getObjectSummaries().iterator();
+				while (objIter.hasNext()) {
+					s3Client.deleteObject(bucketName, objIter.next().getKey());
+				}
+	
+				// If the bucket contains many objects, the listObjects() call
+				// might not return all of the objects in the first listing. Check to
+				// see whether the listing was truncated. If so, retrieve the next page of objects 
+				// and delete them.
+				if (objectListing.isTruncated()) {
+					objectListing = s3Client.listNextBatchOfObjects(objectListing);
+				} else {
+					break;
+				}
+			}
+			//s3Client.deleteBucket(bucketName);
```

Comment:
Why is the bucket no longer deleted?

PS: Perhaps it would be cleaner to create the bucket in a `@BeforeAll` setup method and only delete it in a `@AfterAll` cleanup method instead of creating/deleting it with every single test?
And then reuse the bucket for all tests.
We would need to be sure of course, that bucket-name does not change between tests.

Reply:
Because it no longer has a random name